packages/paste-core/components/datagrid/src/DataGridCell.tsx

```diff
@@ -0,0 +1,88 @@
+import type {BoxElementProps} from '@twilio-paste/box';
+import {useUID} from '@twilio-paste/uid-library';
+import {CompositeItem} from '@twilio-paste/reakit-library';
+import {useMutationObservable} from '@twilio-paste/utils';
+import {DataGridContext} from './DataGridContext';
+import {getAllFocusableIn, toggleActionable, isCell, ensureFocus} from './utils';
+import {Td} from './table/Td';
+
+// This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export
+const isElement = require('lodash.iselement');
+
+export interface DataGridCellProps {
+  element?: BoxElementProps['element'];
+}
+
+export const DataGridCell: React.FC<DataGridCellProps> = ({element = 'DATAGRID_CELL', ...props}) => {
+  // Drastically improves performance https://reakit.io/docs/composite/#performance
+  const id = `datagrid-cell-${useUID()}`;
+  const dataGridState = React.useContext(DataGridContext);
+  const cellRef = React.useRef() as React.MutableRefObject<HTMLTableCellElement>;
+
+  const handleMouseDown = React.useCallback(() => {
+    ensureFocus(cellRef.current);
+  }, []);
+  /**
+   * MutationObserver callback for the cell
+   * Handles correcting tabindex values on the cell and children outside of the React lifecycle
+   * This fixes discrepencies when trying to use Reakit Composite with actionable/navigation modes.
+   */
+  const onListMutation = React.useCallback(
+    (mutationList: MutationRecord[]) => {
+      if (mutationList.length === 0) {
+        return;
+      }
+
+      mutationList.forEach((mutation) => {
+        if (!isElement(mutation.target)) {
+          return;
+        }
+        // Get the element being mutated
+        const target = mutation.target as HTMLElement;
+
+        // Sets the correct tabindex on the cell and child elements
+        // TODO Do we still need this??
+        // toggleActionable(getAllFocusableIn(target), target, dataGridState.actionable);
+
+        /**
+         * Reakit Composite resets tabindex to 0 on cell child focus, which breaks actionable mode on shift-tab.
+         * This fixes the issue by tracking whenever tabindex changes on this cell and setting it to -1
+         * when in actionable mode
+         */
+        if (
+          mutation.attributeName === 'tabindex' &&
+          dataGridState.actionable &&
+          target.tabIndex === 0 &&
+          isCell(target)
+        ) {
+          target.tabIndex = -1;
+        }
+      });
+    },
+    [dataGridState.actionable]
+  );
+  useMutationObservable(cellRef.current, onListMutation);
+
+  /**
+   * When actionable mode changes, toggle the tabindex of the cell and children
+   */
+  React.useEffect(() => {
+    if (cellRef.current) {
+      toggleActionable(getAllFocusableIn(cellRef.current), cellRef.current, dataGridState.actionable);
+    }
+  }, [dataGridState.actionable]);
+
+  return (
+    <CompositeItem
+      {...props}
+      {...dataGridState}
```

Comment:
Is this memoized? If not it might be a candidate for it, do we have any performance info on the new component?

Reply:
This is just context, I'm not sure if / how we should memoize it?